server/src/main/java/com/google/musicanalysis/site/RequestBuilder.java

```diff
@@ -0,0 +1,78 @@
+package com.google.musicanalysis.site;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/*
+    Handles gathering data from the Musixmatch API.
+    This is structured the way it is becuase I will be adding more
+    to each function.
+*/
+public class RequestBuilder {
+
+  // Url building
+  private String baseUrl = "https://api.musixmatch.com/ws/1.1/";
+  private String operation;
+  private String outputReqs = "?format=jsonp&callback=callback&";
+  private String parameters;
+  private final String apiKey = "&apikey=76aaf72ec2a763b4c7aaa7ea135f5ff0";
+
+  private String response;
+
+  /**
+   * Constructor.
+   *
+   * @param operation the API method to send to Musixmatch
+   * @param parameters the parameter string to be sent to Musixmatch
+   */
+  public RequestBuilder(String operation, String parameters) {
+    this.operation = operation;
+    this.parameters = parameters;
+  }
+
+  /** @return the result from Musixmatch API */
+  public String grabResponse() throws MalformedURLException, IOException {
```

Comment:
If you're just returning the string that is retrieved in `getResult()`, then why is this method needed?

Reply:
I put this here to make it easiest on the readers as possible. When they scan through the code, they see that in order to get the response from the Musixmatch API, they instantiate a new class and run obj.grabResponse(), which will return as string. All of the work will remain in private functions to give an "under the hood effect". If they're interested in what's going on, they can look for more info, if not, they can just use it.